docs/grammar/Solidity.g4

```diff
@@ -122,7 +122,7 @@ parameterDeclaration: type=typeName location=dataLocation? name=identifier?;
  * Note that specifying internal or public visibility is deprecated.
  */
 constructorDefinition
-locals[boolean payableSet = false, boolean visibilitySet = false]
```

Comment:
Isn't the type of the parameter defined by antlr? Wouldn't it be better to do the substitution before C++ code generation instead of before creating the Java parser?

Reply:
I was wondering the same as well. I haven't been able to find documentation regarding `boolean` vs `bool`: I think the chosen keyword is bound to the language back-end and since Java is more popular `boolean` is preferred.

Having said that I am fine with doing the substitution before generating C++ code.

Reply:
I think it is better I incorporate this suggestion i.e., do the substitution before generation C++ code.

Closing this PR for that reason in favour of #9675